src/simulation/LoadGenerator.cpp

```diff
@@ -463,52 +492,73 @@ LoadGenerator::handleFailedSubmission(TestAccountPtr sourceAccount,
 }
 
 std::vector<LoadGenerator::TestAccountPtr>
-LoadGenerator::checkAccountSynced(Application& app)
+LoadGenerator::checkAccountSynced(Application& app, bool isCreate)
 {
     std::vector<TestAccountPtr> result;
     for (auto const& acc : mAccounts)
     {
         TestAccountPtr account = acc.second;
-        auto currentSeqNum = account->getLastSequenceNumber();
-        auto reloadRes = loadAccount(account, app);
-        // reload the account
-        if (!reloadRes || currentSeqNum != account->getLastSequenceNumber())
+        auto accountFromDB = *account;
+
+        auto reloadRes = loadAccount(accountFromDB, app);
+        // For account creation, reload accounts from the DB
+        // For payments, ensure that the sequence number matches expected
+        // seqnum. Timeout after 20 ledgers.
+        if (!reloadRes && isCreate)
```

Comment:
code would be easier to read if it was just written with a different nesting:

```c++
if (isCreate)
{
    if (!reloadRes)
    {
          // not created yet
    }
}
else if (!reloadRes)
{
// what should happen here? 
}
else if (account->getLastSequenceNumber() !=
                accountFromDB.getLastSequenceNumber())
{
// not in sync
}
```
note that there is a case that I *think* should not happen (so we should fail the simulation in this case?)